docs/sdk-and-tools/sdk-js/sdk-js-cookbook-v13.md

```diff
@@ -850,6 +850,68 @@ const [sum] = controller.parseQueryResponse(response);
 console.log(sum);
 ```
 
+## Explicit decoding / encoding of values
+
+When needed, you can use the [`BinaryCodec`](https://multiversx.github.io/mx-sdk-js-core/v13/classes/BinaryCodec.html) to [decode and encode values](/developers/data/serialization-overview/) manually.
+
+### Decoding a custom type
+
+Example of decoding a custom type (a structure) called `DepositEvent` from binary data:
+
+```
+import { BinaryCodec } from "@multiversx/sdk-core";
+
+const depositCustomType = abi.getCustomType("DepositEvent");
+const codec = new BinaryCodec();
+let data = Buffer.from("00000000000003db000000", "hex");
+let decoded = codec.decodeTopLevel(data, depositCustomType);
+let decodedValue = decoded.valueOf();
+
+console.log(JSON.stringify(decodedValue, null, 4));
+```
+
+Example of decoding a custom type (a structure) called `Reward` from binary data:
+
+```
+const rewardStructType = abi.getStruct("Reward");
+data = Buffer.from("010000000445474c440000000201f400000000000003e80000000000000000", "hex");
+
+[decoded] = codec.decodeNested(data, rewardStructType);
+decodedValue = decoded.valueOf();
+console.log(JSON.stringify(decodedValue, null, 4));
+```
+
+Example of decoding a custom type (an enum) called `Action` (of [**multisig**](https://github.com/multiversx/mx-contracts-rs/tree/main/contracts/multisig) contract) from binary data:
+
+```
+const actionStructType = abiMultisig.getEnum("Action");
```

Comment:
also instantiate `abiMultisig` here as well. can be instantiated [here](https://github.com/multiversx/mx-docs/blob/v13-codec/docs/sdk-and-tools/sdk-js/sdk-js-cookbook-v13.md?plain=1#L779), as well.

Reply:
Fixed.